lib/app_profiler.rb

```diff
@@ -162,6 +164,14 @@ def profile_sampler_enabled
       false
     end
 
+    def otel_instrumentation_enabled=(value)
+      if value
+        raise ArgumentError,
```

Comment:
I will relax this condition a bit. Rather than raising - will set it to false and warn if otel is not defined.